File: sdks/ruby/lib/nanoservice/logging/log_level.rb
# frozen_string_literal: true

module Nanoservice
  module Logging
    # LogLevel defines the severity levels for structured logging.
    # Levels are ordered: debug < info < warn < error.
    module LogLevel
      DEBUG = :debug
      INFO  = :info
      WARN  = :warn
      ERROR = :error

      ALL = [DEBUG, INFO, WARN, ERROR].freeze

      PRIORITY = {
        debug: 0,
        info:  1,
        warn:  2,
        error: 3
      }.freeze

      # Return the integer priority for a given level symbol.
      # @param level [Symbol] Log level
      # @return [Integer]
      def self.priority(level)
        PRIORITY.fetch(level, 1)
      end

      # Return the uppercase string label for a given level symbol.
      # @param level [Symbol] Log level
      # @return [String]
      def self.label(level)
        level.to_s.upcase
      end

      # Parse a string into a log level symbol.
      # @param str [String] Level string (e.g. "DEBUG", "info")
      # @return [Symbol]
      def self.parse(str)
        case str.to_s.downcase
        when "debug" then DEBUG
        when "warn"  then WARN
        when "error" then ERROR
        else INFO
        end
      end
    end
  end
end

File: sdks/java/src/main/java/com/blok/blok/logging/LogLevel.java
package com.blok.blok.logging;

/**
 * Log severity levels with priority ordering.
 * Higher priority values indicate more severe levels.
 */
public enum LogLevel {

    DEBUG(0),
    INFO(1),
    WARN(2),
    ERROR(3);

    private final int priority;

    LogLevel(int priority) {
        this.priority = priority;
    }

    /**
     * Returns the priority value for this level.
     * Higher values are more severe.
     *
     * @return the priority
     */
    public int getPriority() {
        return priority;
    }

    /**
     * Parses a log level from a string, case-insensitive.
     * Returns the default level if the string is not recognized.
     *
     * @param value        the string to parse
     * @param defaultLevel the fallback level
     * @return the parsed log level
     */
    public static LogLevel fromString(String value, LogLevel defaultLevel) {
        if (value == null || value.isBlank()) {
            return defaultLevel;
        }
        try {
            return LogLevel.valueOf(value.trim().toUpperCase());
        } catch (IllegalArgumentException e) {
            return defaultLevel;
        }
    }
}
